analytical_engine/apps/assortativity/utils.h

```diff
@@ -0,0 +1,23 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef ANALYTICAL_ENGINE_APPS_ASSORTATIVITY_UTILS_H_
+#define ANALYTICAL_ENGINE_APPS_ASSORTATIVITY_UTILS_H_
+
+namespace gs {
+enum class DegreeType { IN, OUT, INANDOUT };
```

Comment:
just put enum to context.h, no need to add a new file

Reply:
DegreeType and some codes about matrix computation can be reused in numeric assortativity app. I will rafactor and put these codes in utils.h